prepare_release.csx

```diff
@@ -0,0 +1,182 @@
+
+using System.Linq;
+
+using Octokit;
+using Octokit.Helpers;
+
+const string repoOwner = "FakeItEasy";
+const string repoName = "FakeItEasy";
+const string releaseFromBranchName = "develop";
+const string targetBranchName = "master";
+
+var version = Args.FirstOrDefault();
+if (version == null)
+{
+    throw new Exception("No version supplied");
+}
+
+var releaseBranchName = $"release/{version}";
+
+static var gitHubClient = GetAuthenticatedGitHubClient();
+
+var existingMilestone = await GetExistingMilestone();
+var existingRelease = await GetExistingRelease();
+var existingIssue = await GetExistingIssue(existingMilestone);
+Milestone nextMilestone;
+if (IsPreRelease(version))
+{
+    nextMilestone = existingMilestone;
+}
+else
+{
+    await RenameMilestone(existingMilestone, version);
+    nextMilestone = await CreateNextMilestone();
+}
+
+await UpdateRelease(existingRelease, version);
+await CreateNextRelease();
+await UpdateIssue(existingIssue, existingMilestone, version);
+await CreateNextIssue(existingIssue, nextMilestone);
+var releaseFromBranch = await GetReleaseFromBranch();
+await CreateReleaseBranch(releaseBranchName, releaseFromBranch);
+await CreatePullRequest(version, releaseBranchName);
+
+public static GitHubClient GetAuthenticatedGitHubClient()
+{
+    var token = ReadGitHubToken();
+    var credentials = new Credentials(token);
+    return new GitHubClient(new ProductHeaderValue("FakeItEasy-build-scripts")) { Credentials = credentials };
+}
+
+public static string ReadGitHubToken()
+{
+    return File.ReadAllText(".githubtoken").Trim();
+}
+
+public static async Task<Milestone> GetExistingMilestone()
+{
+    Console.WriteLine("Fetching milestone 'vNext'...");
+    var milestoneRequest = new MilestoneRequest { State = ItemStateFilter.Open };
+    var existingMilestone = (await gitHubClient.Issue.Milestone.GetAllForRepository(repoOwner, repoName, milestoneRequest))
+        .Single(milestone => milestone.Title == "vNext");
+    Console.WriteLine($"Fetched milestone '{existingMilestone.Title}'");
+    return existingMilestone;
+}
+
+public static async Task<Release> GetExistingRelease()
+{
+    Console.WriteLine("Fetching GitHub release 'vNext'...");
+    var existingRelease = (await gitHubClient.Repository.Release.GetAll(repoOwner, repoName))
+        .Single(release => release.Name == "vNext");
+    Console.WriteLine($"Fetched GitHub release '{existingRelease.Name}'");
+    return existingRelease;
+}
+
+public static async Task<Issue> GetExistingIssue(Milestone existingMilestone)
+{
+    Console.WriteLine("Fetching release issue 'Release vNext'...");
+    var issueRequest = new RepositoryIssueRequest { Milestone = existingMilestone.Number.ToString() };
+    var existingIssue = (await gitHubClient.Issue.GetAllForRepository(repoOwner, repoName, issueRequest))
+        .Single(issue => issue.Title == "Release vNext");

```

Comment:
Oh, just one thing. Shouldn't you filter by open/closed status? This will avoid causing an error if a closed duplicate exists.

Reply:
Default filter is by open status. I can be explicit, if you prefer.

Reply:
I've added such a filter. Sadly, there is nonesuch for Issues. I could explicitly post-filter, which just occurred to me now.